docs/docs/policies.mdx

```diff
@@ -396,6 +398,268 @@ for details.
+`IntentTEDPolicy` helps in surfacing these unique conversation paths from past conversations. For example, if you had
+the following story in your training data:
+
+```yaml
+- story: book_restaurant_table
+  steps:
+  - intent: request_restaurant
+  - action: restaurant_form
+  - active_loop: restaurant_form
+  - action: restaurant_form
+  - active_loop: null
+  - slot_was_set:
+    - requested_slot: null
+```
+
+but an actual conversation might encounter interjections inside the form which you haven't accounted for:
+
+```yaml {10-13}
+- story: actual_conversation
+  steps:
+  - user: |
+        I'm looking for a restaurant.
+    intent: request_restaurant
+  - action: restaurant_form
+  - active_loop: restaurant_form
+  - slot_was_set:
+    - requested_slot: cuisine
+  - user: |
+        Does it matter? I want to be quick.
+    intent: deny
+```
+
+As soon as `deny` intent gets triggered, the policy handling the form will keep requesting for `cuisine` slot
+to be filled, as there is no other way to handle input inside the form according to the training stories.
+Reviewing such conversations are important so that you can account for them in the training data and
+subsequently, the assistant is able to handle them gracefully. `IntentTEDPolicy` can trigger an
+`action_unlikely_intent` action right after `deny` intent which is an indication that this conversation
+has taken a user turn that is not well represented by training stories and needs a review.
+
+To reduce false warnings, `IntentTEDPolicy` has two mechanisms in place at inference time:
+
+1. `IntentTEDPolicy`'s [priority](./policies.mdx#policy-priority) is intentionally kept lower than all
+[rule based policies](./policies.mdx#rule-based-policies). This ensures that if an `action_unlikely_intent`
+is predicted for the last intent, but there is a rule that can be triggered to predict
+the next action, the ensemble of policies is not affected by it and the winning prediction is
+still the one made by `RulePolicy`.
+
+2. `IntentTEDPolicy` does not predict an `action_unlikely_intent` if the last predicted intent
+isn't present in any of the training stories.
+
+As mentioned earlier, while `TEDPolicy` learns the next best action, `IntentTEDPolicy` learns the
+next likely intent that a user can express. Given a conversation context, there can only be a single correct action,
+whereas there can be multiple intents that are likely to occur in the same conversation context. Hence,
+`IntentTEDPolicy` uses a multi-label cross entropy loss function during training.
+
+#### Prediction of `action_unlikely_intent`
+
+After training the internal ML model, `IntentTEDPolicy` does the following for each intent:
+
+1. Collect conversation contexts from training stories for which the corresponding intent is not the correct intent.
+2. Run these conversation contexts through the trained ML model and capture the predicted scores for the corresponding intent.
+3. Compute multiple quantiles
```

Comment:
What about this:


#### Prediction of `action_unlikely_intent`

`IntentTEDPolicy` is invoked immediately after a user utterance and can either predict `action_unlikely_intent` or abstain (in which case other policies will predict actions). To determine if `action_unlikely_intent` should be predicted, `IntentTEDPolicy` computes the confidence score of the intent of the actual user utterance using its internal intent classifier. If the confidence is below a threshold, `IntentTEDPolicy` predicts `action_unlikely_intent`, otherwise it abstains.

Each intent has a unique threshold which is determined at training time after `IntentTEDPolicy`'s internal intent classifier is trained. A synthetic dataset of dialogue contexts paired with unlikely intents is created from your assistant's original training stories. To create the threshold for a specific intent, scores for all examples in the synthetic dataset that have that intent as its label are computed using `IntentTEDPolicy`'s internal intent classifier 
and the threshold is set to be the maximum of these confidence scores. In its default setting, `IntentTEDPolicy` would predict an `action_unlikely_intent` for any of the examples in the synthetic dataset.

If the `IntentTEDPolicy` is too sensitive (predicting too many `action_unlikely_intent`s) its tolerance parameter can be raised to decrease the chances of predicting `action_unlikely_intent`. The tolerance parameter defaults to `0.0` and ranges between `0.0` and `1.0` inclusive. At a tolerance of `0.25`, `IntentTEDPolicy` would not predict an `action_unlikely_intent` on 25% of the examples from the synthetic dataset. When the tolerance is set to `1.0`, 100% of the examples would not trigger an `action_unlikely_intent`.

Reply:
Yeah I really like this! My only other suggestion would be to change the middle paragraph slightly:

This threshold score is learned at training time but can be adjusted at test time via the `tolerance` parameter.  The higher the `tolerance`, the lower the intent's score (the more unlikely the intent) needs to be before IntentTED triggers the `action_unlikely_intent`.

Reply:
Thank you both! These look really good 🚀

Reply:
I used the first two paragraphs for this section and moved the 3rd paragraph to the description of `tolerance` parameter in the `Configuration` section connecting the two.